File: lab01.py
import pandas as pd

olist = pd.read_parquet(
    "https://github.com/padsInsper/202533-padsv/releases/download/dados/olist_items.parquet"
)

pagamento_counts = olist['types'].value_counts().reset_index()
pagamento_counts.columns = ['Forma de pagamento', 'Quantidade']
pagamento_filtro = pagamento_counts[pagamento_counts['Quantidade'] > 100]

pagamento_filtro['Quantidade (milhares)'] = round(pagamento_filtro['Quantidade'] / 1000, 2)

pagamento_filtro['n'] = pagamento_filtro['Quantidade (milhares)']

from plotnine import *

(
    ggplot(pagamento_filtro, aes(y = 'Quantidade (milhares)', x = 'Forma de pagamento'))
    + geom_col()
    + coord_flip()
    + geom_label(aes(label = 'round(n, 2)', y = 'n / 2'))
)

File: lab01.R
olist <- arrow::read_parquet(
    "https://github.com/padsInsper/202533-padsv/releases/download/dados/olist_items.parquet"
)

dplyr::glimpse(olist)

# types

dados_grafico <- olist |>
    dplyr::count(types) |>
    dplyr::filter(n > 100) |>
    dplyr::mutate(n = n / 1000)

library(ggplot2)

dados_grafico |>
    ggplot(aes(x = n, y = types)) +
    geom_col(fill = "#8ae3d7", width = .5) +
    geom_label(aes(label = round(n, 2), x = n / 2)) +
    theme_dark(16) +
    labs(
        x = "Quantidade\n(milhares)",
        y = "Forma de pagamento",
        title = "Formas de pagamento mais comuns",
        subtitle = "Considerando tipos com mais de 100 observações",
        caption = "Fonte: Olist"
    ) +
    theme(
        panel.background = element_rect(fill = "gray20"),
        plot.background = element_rect(fill = "gray10"),
        text = element_text(family = "serif", colour = "white"),
        axis.text = element_text(family = "serif", colour = "white")
    )


ggplot()
